src/VirtualClient/VirtualClient.Contracts.UnitTests/Parser/TextParserExtensionsTests.cs

```diff
@@ -34,17 +35,84 @@ public void TextParserExtensionsTranslateByteUnitAsExpected(string orginalText,
         [TestCase("100tb", MetricUnit.Gigabytes, "102400")]
         [TestCase("100pb", MetricUnit.Terabytes, "102400")]
         [TestCase("1tb", MetricUnit.Kilobytes, "1073741824")]
-        public void TextParserExtensionsTranslateStorageByUnitAsExpected(string orginalText, string metricUnit, string expectedOutput)
+        public void TextParserExtensionsTranslateStorageByUnitAsExpected(string originalText, string metricUnit, string expectedOutput)
         {
-            Assert.IsTrue(string.Equals(TextParsingExtensions.TranslateStorageByUnit(orginalText, metricUnit), expectedOutput));
+            Assert.IsTrue(string.Equals(TextParsingExtensions.TranslateStorageByUnit(originalText, metricUnit), expectedOutput));
         }
 
         [Test]
         [TestCase("100k", "100000")]
         [TestCase("100m", "100000000")]
-        public void TextParserExtensionsTranslateNumericUnitAsExpected(string orginalText, string expectedOutput)
+        public void TextParserExtensionsTranslateNumericUnitAsExpected(string originalText, string expectedOutput)
         {
-            Assert.IsTrue(string.Equals(TextParsingExtensions.TranslateNumericUnit(orginalText), expectedOutput));
+            Assert.IsTrue(string.Equals(TextParsingExtensions.TranslateNumericUnit(originalText), expectedOutput));
+        }
+
+        [Test]
+        [TestCase("0.0h", "0")]
+        [TestCase("1h", "3600")]
+        [TestCase("1hrs", "3600")]
+        [TestCase("1hr", "3600")]
+        [TestCase("1.0hours", "3600")]
+        [TestCase("0.01HOUR", "36")]
+        //
+        [TestCase("1m", "60")]
+        [TestCase("1minutes", "60")]
+        [TestCase("1MINUTE", "60")]
+        [TestCase("0.5MINUTE", "30")]
+        //
+        [TestCase("1000ms", "1")]
+        [TestCase("1000milliseconds", "1")]
+        [TestCase("1000millisecond", "1")]
+        [TestCase("0.99MILLISECONDS", "0.00099")]
+        [TestCase("0.99MiLLiseconDS", "0.00099")]
+        //
+        [TestCase("1000000us", "1")]
+        [TestCase("1000000microsecond", "1")]
+        [TestCase("1000000microseconds", "1")]
+        //
+        [TestCase("1000000000.00ns", "1")]
+        [TestCase("1000000000nanosecond", "1")]
+        [TestCase("1000000000nanoseconds", "1")]
+        //
+        [TestCase("1s", "1")]
+        [TestCase("1second", "1")]
+        [TestCase("2seconds", "2")]
+        public void TextParserExtensionsTranslateTimeUnitToSecondAsExpected(string originalText, string expectedOutput)
+        {
+            string result = TextParsingExtensions.TranslateToSecondUnit(originalText);
+            Assert.AreEqual(result, expectedOutput);
+        }
+
+        [Test]
+        [TestCase(" -2seconds ")]
+        [TestCase("-1minute")]
+        public void TextParserExtensionsDoesNotSupportNegativeTime(string originalText)
```

Comment:
Just curious, why we don't support negatives? They could be valid output from workload

Reply:
I have not experienced need for negative time in Virtual Client yet. 
Not supporting negative time (for now) to stay consistent with Byte function.